devices/uart-nrf9160/uart.c

```diff
@@ -0,0 +1,323 @@
+/*
+ * Phoenix-RTOS
+ *
+ * plo - operating system loader
+ *
+ * nRF9160 UART driver
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Damian Loewnau
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+#include <lib/errno.h>
+#include <lib/lib.h>
+#include <devices/devs.h>
+
+#include <board_config.h>
+
+#define BUFFER_SIZE 0x20
+
+typedef struct {
+	volatile unsigned int *base;
+	unsigned int cnt;
+	unsigned int irq;
+
+	volatile char *tx_dma;
+	volatile char *rx_dma;
+
+	u8 dataRx[BUFFER_SIZE];
+	cbuffer_t cbuffRx;
+
+	u8 dataTx[BUFFER_SIZE];
+	cbuffer_t cbuffTx;
+} uart_t;
+
+
+struct {
+	uart_t uarts[UART_MAX_CNT];
+} uart_common;
+
+/* clang-format off */
+enum { cr1 = 0, cr2, cr3, brr, gtpr, rtor, rqr, isr, icr, rdr, tdr };
+/* clang-format on */
+
+static int uartLut[] = { UART0, UART1, UART2, UART3 };
+
+
+static const struct {
+	void *base;
+	unsigned int irq;
+	unsigned int baudrate;
+	volatile char *tx_dma;
+	volatile char *rx_dma;
+	unsigned char txpin;
+	unsigned char rxpin;
+	unsigned char rtspin;
+	unsigned char ctspin;
+} uartInfo[] = {
+	{ (void *)UART0_BASE, UART0_IRQ, UART_BAUDRATE, (volatile char *)UART0_TX_DMA, (volatile char *)UART0_RX_DMA,
+		UART0_TX, UART0_RX, UART0_RTS, UART0_CTS },
+	{ (void *)UART1_BASE, UART1_IRQ, UART_BAUDRATE, (volatile char *)UART1_TX_DMA, (volatile char *)UART1_RX_DMA,
+		UART1_TX, UART1_RX, UART1_RTS, UART1_CTS },
+	{ (void *)UART2_BASE, UART2_IRQ, UART_BAUDRATE, (volatile char *)UART2_TX_DMA, (volatile char *)UART2_RX_DMA,
+		UART2_TX, UART2_RX, UART2_RTS, UART2_CTS },
+	{ (void *)UART3_BASE, UART3_IRQ, UART_BAUDRATE, (volatile char *)UART3_TX_DMA, (volatile char *)UART3_RX_DMA,
+		UART3_TX, UART3_RX, UART3_RTS, UART3_CTS }
```

Comment:
Cast via `void *` is enough
```suggestion
	{ (void *)UART0_BASE, UART0_IRQ, UART_BAUDRATE, (void *)UART0_TX_DMA, (void *)UART0_RX_DMA,
		UART0_TX, UART0_RX, UART0_RTS, UART0_CTS },
	{ (void *)UART1_BASE, UART1_IRQ, UART_BAUDRATE, (void *)UART1_TX_DMA, (void *)UART1_RX_DMA,
		UART1_TX, UART1_RX, UART1_RTS, UART1_CTS },
	{ (void *)UART2_BASE, UART2_IRQ, UART_BAUDRATE, (void *)UART2_TX_DMA, (void *)UART2_RX_DMA,
		UART2_TX, UART2_RX, UART2_RTS, UART2_CTS },
	{ (void *)UART3_BASE, UART3_IRQ, UART_BAUDRATE, (void *)UART3_TX_DMA, (void *)UART3_RX_DMA,
		UART3_TX, UART3_RX, UART3_RTS, UART3_CTS }
```